libcasr/src/csharp.rs

```diff
@@ -0,0 +1,367 @@
+        assert_eq!(exceptions[1].short_description, "System.IO.IOException");
+        assert_eq!(exceptions[1].description, "cccc");
+        assert_eq!(exceptions[2].short_description, "System.IO.IOException");
+        assert_eq!(exceptions[2].description, "cccc\n\n");
+        assert_eq!(exceptions[3].short_description, "System.ArgumentException");
+        assert_eq!(
+            exceptions[3].description,
+            "1111 ---> System.IO.IOException: cccc\n\n\n"
+        );
+    }
+
+    #[test]
+    fn test_csharp_stacktrace() {
+        let stream = "Unhandled exception. System.ArithmeticException: 123
+
+321
+
+7
+
+ ---> System.ArgumentException: 1111
+ ---> System.IO.IOException: cccc
+ ---> System.IO.IOException: bbbb
+ ---> System.IO.IOException: aaaa
+ ---> System.IO.IOException: I/O error occurred.
+   --- End of inner exception stack trace ---
+   --- End of inner exception stack trace ---
+   --- End of inner exception stack trace ---
+
+
+   at C.qwe()
+   at B..ctor() in /home/user/dotnet/2/A.cs:line 37
+   at A`1.<>c.<set_Q>b__1_1() in /home/user/dotnet/2/A.cs:line 15
+
+   at A`1.h[Z](Func`1 a)
+   --- End of inner exception stack trace ---
+   --- End of inner exception stack trace ---
+  at A`1[T].<set_Q>g__g|1_0 (System.Int32[] arr) <0x40b745f0 + 0x00122> in /home/user/mono/2/src/2.cs:13
+  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <c79446e93efd45a0b7bc2f9631593aff>:0
+
+  at A`1[T].set_Q (System.Int32 value) <0x40275140 + 0x00082> in <f6b2b0ea894844dc83a96f9504d8f570#610bc057486c618efb3936233b088988>:0
+  at (wrapper runtime-invoke) staticperformanceoptimization.runtime_invoke_void (object,intptr,intptr,intptr) <0xffffffff>
+
+
+  at (wrapper managed-to-native) System.Drawing.GDIPlus:GdiplusStartup (ulong&,System.Drawing.GdiplusStartupInput&,System.Drawing.GdiplusStartupOutput&)
+  at Program+<>c.<Main>b__0_2 (System.Int32 i) <0x7f1c08e516b0 + 0x00035> in <f6b2b0ea894844dc83a96f9504d8f570#17898f75ae7b737216569c39168a3967>:0
+  at Program.<Main>g__f|0_0 (System.Func`2[T,TResult] d, System.Int32& l, System.Int32 m) <0x7f1c08e51140 + 0x000d9> in <f6b2b0ea894844dc83a96f9504d8f570#17898f75ae7b737216569c39168a3967>:0
+  at Program.Main () <0x7f1c08e51010 + 0x000ea> in <f6b2b0ea894844dc83a96f9504d8f570#17898f75ae7b737216569c39168a3967>:0
+";
+
+        let trace = [
+            "--- End of inner exception stack trace ---",
+            "--- End of inner exception stack trace ---",
+            "--- End of inner exception stack trace ---",
+            "at C.qwe()",
+            "at B..ctor() in /home/user/dotnet/2/A.cs:line 37",
+            "at A`1.<>c.<set_Q>b__1_1() in /home/user/dotnet/2/A.cs:line 15",
+            "at A`1.h[Z](Func`1 a)",
+            "--- End of inner exception stack trace ---",
+            "--- End of inner exception stack trace ---",
+            "at A`1[T].<set_Q>g__g|1_0 (System.Int32[] arr) <0x40b745f0 + 0x00122> in /home/user/mono/2/src/2.cs:13",
```

Comment:
I think, stactrace starts from here, isn't it? I think,  everything above should be removed from stacktrace array.

Reply:
@BMikh0, you may take a look at Java stack trace parsing code. It handles situation with multiple stack traces and chooses the right one.

Reply:
Stacktrace in C# consists of multiple pieces, but they form the whole stacktace together. They don't repeat as in Java stacktraces.